src/types/context/app_context.go

```diff
@@ -24,8 +24,8 @@ func NewAppContext(location string, config types.AppConfig) (*AppContext, error)
 	return appContext, appContext.initializeServiceContexts()
 }
 
-// GetServiceContextByLocation returns the service context for the given location
-func (a *AppContext) GetServiceContextByLocation(location string) *ServiceContext {
+// GetServiceContext returns the service context for the given location
```

Comment:
I see that we're getting service context by role in the private method `AppContext.getServiceContext()` (which perhaps we can rename to `getServiceContextByRole`, but where else are we doing that? It seems like `GetAppContext` uses location, so it would make sense that `GetServiceContext` does as well. I'm just trying to make the public functions as simple and sensible as possible.

Reply:
`AppContext` has a `ServiceContexts` property which is a map from role to `ServiceContext`